finality-aleph/src/network/manager/discovery.rs

```diff
@@ -129,9 +143,9 @@ impl Discovery {
     /// that we should send as a result of it.
     pub fn handle_message(
         &mut self,
-        message: DiscoveryMessage,
-        handler: &mut SessionHandler,
-    ) -> (Vec<Multiaddr>, Vec<DiscoveryCommand>) {
+        message: DiscoveryMessage<M>,
+        handler: &mut SessionHandler<M>,
+    ) -> (Vec<M>, Vec<DiscoveryCommand<M>>) {
```

Comment:
Same thing about tuples: here it's a public API so I'm even more convinced :smile: